stackslib/src/chainstate/nakamoto/tests/mod.rs

```diff
@@ -1636,3 +1643,272 @@ pub fn test_get_highest_nakamoto_tenure() {
     assert_eq!(highest_tenure.tenure_index, 3);
     assert_eq!(highest_tenure.num_blocks_confirmed, 10);
 }
+
+/// Test that we can generate a .miners stackerdb config.
+/// The config must be stable across sortitions -- if a miner is given slot i, then it continues
+/// to have slot i in subsequent sortitions.
+#[test]
+fn test_make_miners_stackerdb_config() {
+    let test_signers = TestSigners::default();
+    let mut peer = boot_nakamoto(
+        function_name!(),
+        vec![],
+        test_signers.aggregate_public_key.clone(),
+    );
+
+    let miner_keys: Vec<_> = (3..13).map(|_| StacksPrivateKey::new()).collect();
+    let miner_hash160s: Vec<_> = miner_keys
+        .iter()
+        .map(|miner_privkey| {
+            let miner_pubkey = StacksPublicKey::from_private(miner_privkey);
+            let miner_hash160 = Hash160::from_node_public_key(&miner_pubkey);
+            miner_hash160
+        })
+        .collect();
+    let miner_addrs: Vec<_> = miner_hash160s
+        .iter()
+        .map(|miner_hash160| StacksAddress {
+            version: 1,
+            bytes: miner_hash160.clone(),
+        })
+        .collect();
+
+    debug!("miners = {:#?}", &miner_hash160s);
+
+    // extract chainstate and sortdb -- we don't need the peer anymore
+    let chainstate = &mut peer.stacks_node.as_mut().unwrap().chainstate;
+    let sort_db = peer.sortdb.as_mut().unwrap();
+    let mut last_snapshot = SortitionDB::get_canonical_burn_chain_tip(sort_db.conn()).unwrap();
+
+    // make two leader keys (miner 1 and miner 2)
+    let mut miners = vec![];
+    for i in 3..13 {
+        let vrf_privkey = VRFPrivateKey::new();
+        let vrf_pubkey = VRFPublicKey::from_private(&vrf_privkey);
+        let miner = LeaderKeyRegisterOp {
+            consensus_hash: last_snapshot.consensus_hash.clone(),
+            public_key: vrf_pubkey,
+            memo: miner_hash160s[i - 3].0.to_vec(),
+            txid: Txid([i as u8; 32]),
+            vtxindex: 1 + (i as u32),
+            block_height: last_snapshot.block_height + 1,
+            burn_header_hash: last_snapshot.burn_header_hash.clone(),
+        };
+        miners.push(miner);
+    }
+
+    let mut stackerdb_configs = vec![];
+    let mut stackerdb_chunks = vec![];
+
+    // synthesize some sortitions and corresponding winning block-commits
+    for i in 3..13 {
+        // no winner every 3rd sortition
+        let sortition = i > 3 && i % 3 != 0;
```

Comment:
I don't understand these indexes. Why not `0..10`?

Reply:
Will update to 0..10 :)